php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -104,11 +104,12 @@ class NDB_BVL_Battery
      * @param integer $center_ID    The center of the candidate having a battery
      *                              created
      * @param boolean $firstVisit   Whether this is a first visit battery.
-     *                              true means ONLY first visit
-     *                              false means NOT first visit
-     *                              null can be any visit.
+     *                              true means ONLY first visit false
+     *                              means NOT first visit null can be any
```

Comment:
This param should probably be `boolean|null` then, even though that feels horrible.

Reply:
@johnsaigle Null is equivalent to false for a boolean and so I don't think that's correct.

Reply:
I don;t think that's correct. `true` `false` and `null` are all different things in PHP. The function comment here makes it sound like 3 different code paths are taken depending on the value of this param.

Reply:
`null` is "falsey" but `null === false` is not a correct statement.

Reply:
@johnsaigle nice I'll add the boolean|null.

Reply:
Also maybe add a FIXME comment that this code should be refactored. This is a bad code smell